src/directives/internal/class.js

```diff
@@ -50,8 +60,19 @@ export default {
       var i = this.prevKeys.length
       while (i--) {
         var key = this.prevKeys[i]
-        if (key && (!value || !contains(value, key))) {
+        if (key && (!value || !contains(value, key)) && !isPlainObject(key)) {
           removeClass(this.el, key)
+        } else if (key && isPlainObject(key)) {
+          var keys = Object.keys(key)
+          for (var k = 0; k < keys.length; k++) {
+            for (var j = 0; j < value.length; j++) {
```

Comment:
`value` is not guaranteed to be an Array - it can be an Object too. I think it's better to improve `contains` so that it also checks objects inside arrays.